api/src/main/java/org/openmrs/api/db/hibernate/HibernateUserDAO.java

```diff
@@ -118,16 +119,18 @@ public User getUserByEmail(String email) {
 	}
 	
 	/**
-	 * @see org.openmrs.api.UserService#getUserByActivationKey(java.lang.String)
+	 * @see org.openmrs.api.UserService#getLoginCredentialByToken(java.lang.String)
 	 */
 	@Override
 	@SuppressWarnings("unchecked")
-	public User getUserByActivationKey(String activationKey) {
-		User user = (User) sessionFactory.getCurrentSession().createCriteria(User.class).add(Restrictions.like("activationKey", activationKey, MatchMode.START)).uniqueResult();	
-		if(user != null) {
-			String[] tokens = user.getActivationKey().split(":");
-			if(tokens[0].equals(activationKey)) {
-				return user;
+	public LoginCredential getLoginCredentialByToken(String token) {
+		String key = Security.encodeString(token);
+		LoginCredential loginCred = (LoginCredential) sessionFactory.getCurrentSession().createCriteria(LoginCredential.class).add(Restrictions.like("activationKey", key, MatchMode.START)).uniqueResult();	
+		if(loginCred != null) {
+			String[] credTokens = loginCred.getActivationKey().split(":");
+			// if(credTokens[0].equals(key) && (System.currentTimeMillis() <= Long.parseLong(credTokens[1]) ))
```

Comment:
Don't leave commented out code behind please

Reply:
ok will fix it i disabled it because of testing purpose